src/s_inter.c

```diff
@@ -393,6 +392,11 @@ void sys_set_priority(int mode)
 
 /* ------------------ receiving incoming messages over sockets ------------- */
 
+unsigned char *sys_getrecvbuf(void)
+{
+    return INTER->i_recvbuf;
+}
+
```

Comment:
what i personally don't like so much is, that the size of the buffer is hardcoded somewhere else (in `s_net.h`).
couldn't that be returned along with the buffer (which should make it more robust for the concocted situtation where `s_inter.c` has been compiled with a different value for `NET_MAXPACKETSIZE` than `x_net.c`):

~~~C
unsigned char *sys_getrecvbuf(unsigned int*size)
{
    if(size)*size=NET_MAXPACKETSIZE;
    return INTER->i_recvbuf;
}
~~~

it might be overengineering though.

Reply:
> which should make it more robust for the concocted situtation where s_inter.c has been compiled with a different value for NET_MAXPACKETSIZE than x_net.c):

In that case the build system would be horribly broken.

> what i personally don't like so much is, that the size of the buffer is hardcoded somewhere else (in s_net.h).

I think for internal use this is no problem.

*But* I'm just asking myself whether we should (semi) export the receive buffer, e.g. in `s_net.h? I guess this could be useful for networking externals. In this case we really have to provide the size, because the external might be compiled against a different version of `s_net.h`. What do you think?

Reply:
sag ich ja :+1: